libs/libc/stdlib/lib_rand48.c

```diff
@@ -0,0 +1,194 @@
+
+/****************************************************************************
+ * Name: srand48
+ ****************************************************************************/
+
+void srand48(long seed)
+{
+  unsigned short p[3];
+
+  p[0] = 0x330e;
+  p[1] = seed;
+  p[2] = seed >> 16;
+  seed48(p);
+}
+
+/****************************************************************************
+ * Name: seed48
+ ****************************************************************************/
+
+FAR unsigned short *seed48(unsigned short seed16v[3])
+{
+  static unsigned short p[3];
+
+  memcpy(p, g_seed48, sizeof p);
+  memcpy(g_seed48, seed16v, sizeof p);
+  return p;
+}
+
+/****************************************************************************
+ * Name: lcong48
+ ****************************************************************************/
+
+void lcong48(unsigned short p[7])
+{
+  memcpy(g_seed48, p, sizeof(g_seed48));
+}
+
+/****************************************************************************
+ * Name: jrand48
+ *
+ * Description:
+ *   Return signed long integers uniformly distributed over the
+ *   interval [-2^31, 2^31).
+ *
+ ****************************************************************************/
+
+long jrand48(unsigned short s[3])
+{
+  return (long)(rand48_step(s, g_seed48 + 3) >> 16);
+}
+
+/****************************************************************************
+ * Name: mrand48
+ *
+ * Description:
+ *   Return signed long integers uniformly distributed over the
+ *   interval [-2^31, 2^31).
+ *
+ ****************************************************************************/
+
+long mrand48(void)
+{
+  return jrand48(g_seed48);
+}
+
+/****************************************************************************
+ * Name: nrand48
+ *
+ * Description:
+ *   Return nonnegative long integers  uniformly  distributed over the
+ *   interval [0, 2^31).
+ *
+ ****************************************************************************/
+
+long nrand48(unsigned short s[3])
+{
+  return rand48_step(s, g_seed48 + 3) >> 17;
+}
+
+/****************************************************************************
+ * Name: lrand48
+ *
+ * Description:
+ *   Return nonnegative long integers  uniformly  distributed over the
+ *   interval [0, 2^31).
+ *
+ ****************************************************************************/
+
+long lrand48(void)
+{
+  return nrand48(g_seed48);
+}
+
+/****************************************************************************
+ * Name: erand48
+ *
+ * Description:
+ *   Return nonnegative double-precision floating-point values uniformly
+ *   distributed over the interval [0.0, 1.0).
+ *
+ ****************************************************************************/
+
+double erand48(unsigned short s[3])
+{
+  union
+    {
+      uint64_t u;
+      double f;
+    } x;
+
+  x.u = 0x3ff0000000000000ull | rand48_step(s, g_seed48 + 3) << 4;
```

Comment:
Should this be guarded by `#ifdef CONFIG_HAVE_LONG_LONG` and `#ifdef CONFIG_HAVE_DOUBLE`?

Reply:
Done!